scripts/lib-franklin.js

```diff
@@ -316,6 +316,49 @@ export function readBlockConfig(block) {
   return config;
 }
 
+/**
+ * adds animation on metadata level
+ */
+export function addAnimation(animationTypeMeta, root) {
+  const addClass = (elem, className) => {
+    elem.classList.add(className);
+  };
+
+  const elementsToAnimate = root.querySelectorAll('h1, h2, h3, h4, h5, h6, a, img');
+
+  const applyAnimation = (entries, observer) => {
+    entries.forEach((entry) => {
+      if (entry.isIntersecting && entry.intersectionRatio >= 0.3) {
+        addClass(entry.target, `animated-${animationTypeMeta}`);
+
+        observer.unobserve(entry.target); // Stop observing once animation is applied
+      }
+    });
+  };
+
+  const observer = new IntersectionObserver(applyAnimation, { threshold: 0.3 });
```

Comment:
looking at the code I had an additional thought: if we overwrite the animation in a section don't we have to remove all those listeners?